frame/example-offchain-worker/src/tests.rs

```diff
@@ -71,22 +78,37 @@ impl frame_system::Trait for Test {
 }
 
 type Extrinsic = TestXt<Call<Test>, ()>;
-type SubmitTransaction = frame_system::offchain::TransactionSubmitter<
-	crypto::Public,
-	Test,
-	Extrinsic
->;
-
-impl frame_system::offchain::CreateTransaction<Test, Extrinsic> for Test {
-	type Public = sp_core::sr25519::Public;
-	type Signature = sp_core::sr25519::Signature;
-
-	fn create_transaction<F: frame_system::offchain::Signer<Self::Public, Self::Signature>>(
-		call: <Extrinsic as ExtrinsicsT>::Call,
-		_public: Self::Public,
-		_account: <Test as frame_system::Trait>::AccountId,
-		nonce: <Test as frame_system::Trait>::Index,
-	) -> Option<(<Extrinsic as ExtrinsicsT>::Call, <Extrinsic as ExtrinsicsT>::SignaturePayload)> {
+type AccountId = <<Signature as Verify>::Signer as IdentifyAccount>::AccountId;
```

Comment:
So this `Signature` is coming from sp_core::crypto directly. Not sure if we want to showcase this as an example? ain't this an anti-pattern and the mock file should create its own `app_crypto` and use it.

(don't quote me on this :D)

Reply:
?

Reply:
Signature actually comes from:
https://github.com/paritytech/substrate/blob/8c1858048148c1c717123b0e4d360f8ad1be5bf2/bin/node/primitives/src/lib.rs#L31

Reply:
Sorry, my previous comment is wrong. I think the point of doing this is to have acces to a type that implements `IdentifyAccount`. The `Public` type generated by `app_crypto` does not implement this directly. `IdentifyAccount` is implemented for the generic `{sr/ed}25519::Public` types.

Reply:
Indeed, you need `IdentifyAccount `. But my point is, that a key type must be used here that resembles that most with the realistic scenario. And I am not sure if the core, untyped sr25519 is the way to go. Maybe you should try and mimic what is happening in `node/runtime`?

Reply:
`AccountId` is never supposed to be `app_crypto`. This is a runtime-wide account id, app-crypto is used for pallet-specific cases.

In `node/runtime` we use `MultiSigner/MultiSignature`, but using a single crypto `sr25519` is totally fine.